rdsbroker/broker.go

```diff
@@ -391,10 +391,57 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
+		} else {
+			dbName = b.dbName(dbInstanceDetails.Identifier)
+		}
+
+		sqlEngine, err := b.sqlProvider.GetSQLEngine(dbInstanceDetails.Engine)
+		if err != nil {
+			b.logger.Error("Could not get the SQL Engine", err)
+			return err
+		}
+
+		err = sqlEngine.Open(dbAddress, dbPort, dbName, masterUsername, b.masterPassword(serviceInstanceID))
+		if err != nil {
```

Comment:
Do we want to change password for any type of error, even if it's network error?
lib/pq implements connection here:
https://github.com/lib/pq/blob/165a3529e799da61ab10faed1fabff3662d6193f/conn.go#L347  so you can check the type.